app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -133,29 +108,26 @@ class SpotlightFragment : InjectableFragment(), SpotlightNavigationListener, Spo
         spotlightTarget.feature
       ).toLiveData()
 
-    // use activity as observer because this fragment's view hasn't been created yet.
+    // Use activity as observer because this fragment's view hasn't been created yet.
     featureViewStateLiveData.observe(
       activity,
       object : Observer<AsyncResult<SpotlightViewState>> {
         override fun onChanged(it: AsyncResult<SpotlightViewState>?) {
           if (it is AsyncResult.Success) {
-            val viewState = (it.value)
-            if (viewState == SpotlightViewState.SPOTLIGHT_SEEN) {
-              return
-            } else if (viewState == SpotlightViewState.SPOTLIGHT_NOT_SEEN) {
-              createTarget(spotlightTarget)
-              featureViewStateLiveData.removeObserver(this)
-            }
+            if (it.value == SpotlightViewState.SPOTLIGHT_SEEN) return
+            createTarget(spotlightTarget)
```

Comment:
```suggestion
            if (it.value == SpotlightViewState.SPOTLIGHT_NOT_SEEN) {
              createTarget(spotlightTarget)
            }
```

Otherwise, won't the observer stay around forever if it's been seen?

Reply:
done